Reorganize performance articles.

stbof opened the pull request:
See description in https://cockroachlabs.atlassian.net/browse/DOC-2576.

- Move rule application tutorial to new page.
- Adds missing links to troubleshooting articles.
- Consistent titles in TOC and pages.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/13026)
<!-- Reviewable:end -->

kevin-v-ngo reviewed (approved):
Thanks for reorganizing the performance content @stbof! Don't think we need to do this in this PR but we should consider merging Apply Statement Performance Rules into Performance Best Practices at some point.

kevin-v-ngo commented:
> Thanks for reorganizing the performance content @stbof! Don't think we need to do this in this PR but we should consider merging Apply Statement Performance Rules into Performance Best Practices at some point.

Or perhaps "Apply SQL Statement Performance Rules" moved to a Tutorials section

stbof commented:
> > Thanks for reorganizing the performance content @stbof! Don't think we need to do this in this PR but we should consider merging Apply Statement Performance Rules into Performance Best Practices at some point.
> 
> Or perhaps "Apply SQL Statement Performance Rules" moved to a Tutorials section

I agree Tutorials makes more sense than Best Practices, because the structure of the Apply page is a tutorial. However, do you mean a Tutorials section within Performance?

jseldess commented:
@stbof, whenever you rename a page, please be sure to add a server-side redirect in `_redirects` in addition to updating all relevant internal links. 





<!-- Sent from Reviewable.io -->

stbof commented:
> @stbof, whenever you rename a page, please be sure to add a server-side redirect in `_redirects` in addition to updating all relevant internal links.

I didn't rename anything. I moved them around in the TOC and added a new page.

jseldess commented on _includes/v21.2/sidebar-data/develop.json:
```diff
@@ -274,22 +274,64 @@
-          "title": "Use the EXPLAIN Statement",
+          "title": "Apply Statement Performance Rules",
           "urls": [
-            "/${VERSION}/sql-tuning-with-explain.html"
```
@stbof, I think I misinterpreted this as a renaming. The old page is still around, just not listed here in the sidenav?

jseldess replied:
Nevermind. I see it listed just above. Sorry for not looking more closely.

rytaft reviewed (commented):
> Reviewed 24 of 30 files at r1, 4 of 5 files at r2, 1 of 1 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13026)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jordanlewis, @stbof, and @vy-ton)

___
*[v21.2/apply-statement-performance-rules.md, line 24 at r4](https://reviewable.io/reviews/cockroachdb/docs/13026#-Mwhg2i14Xhaz4VKiVLq:-Mwhg2i14Xhaz4VKiVLr:bnyu08n) ([raw file](https://github.com/cockroachdb/docs/blob/91a8193c2fd3d938a318298e995c0a315b581745/v21.2/apply-statement-performance-rules.md#L24)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ shell
> cockroach sql --insecure
> ```

I thought you were just using `cockroach demo` (at least that's what's implied below)?
___
*[v21.2/apply-statement-performance-rules.md, line 109 at r4](https://reviewable.io/reviews/cockroachdb/docs/13026#-MwhfcOwCdsLxKPsyWY-:-MwhfcOx0VU_zuI23jCp:b-yhfafa) ([raw file](https://github.com/cockroachdb/docs/blob/91a8193c2fd3d938a318298e995c0a315b581745/v21.2/apply-statement-performance-rules.md#L109)):*
> ```Markdown
> There is a `rider_id` field that you can use to match each ride to a user. There is also a `start_time` field that you can use to filter the rides by date.
> 
> This means that to get the information you want, you'll need to do a [join][joins] on the `users` and `rides` tables.
> ```

this [join] link looks like it might not work
___
*[v21.2/performance-features-overview.md, line 12 at r4](https://reviewable.io/reviews/cockroachdb/docs/13026#-MwhhyBw7yrUpLWHpymX:-MwhhyBw7yrUpLWHpymY:bvoj1wj) ([raw file](https://github.com/cockroachdb/docs/blob/91a8193c2fd3d938a318298e995c0a315b581745/v21.2/performance-features-overview.md#L12)):*
> ```Markdown
> [Indexes](indexes.html) improve query performance by helping CockroachDB locate data without having to look through every row of a table. You typically add and tune indexes when [designing your database schema](schema-design-overview.html#indexes).
> 
> The following features are automatically configured by CockroachDB and employed during query planning and execution. However, you can change the configuration or disable the feature:
> ```

nit: you can't disable the optimizer anymore. I think you can disable the other two, though.
___
*[v21.2/query-behavior-troubleshooting.md, line 75 at r4](https://reviewable.io/reviews/cockroachdb/docs/13026#-MwhiNLK-yUjhHCKjxd0:-MwhiNLK-yUjhHCKjxd1:b-h1eo9h) ([raw file](https://github.com/cockroachdb/docs/blob/91a8193c2fd3d938a318298e995c0a315b581745/v21.2/query-behavior-troubleshooting.md#L75)):*
> ```Markdown
> You can also use an [`EXPLAIN ANALYZE`](explain-analyze.html) statement, which executes a SQL query and returns a physical query plan with execution statistics. Query plans can be used to troubleshoot slow queries by indicating where time is being spent, how long a processor (i.e., a component that takes streams of input rows and processes them according to a specification) is not doing work, etc.
> 
> Cockroach Labs recommends sending either the diagnostics bundle or the `EXPLAIN ANALYZE` output to our [support team](support-resources.html) for analysis.
> ```

we always prefer the bundle if it's available (not sure if you need to say that, but FYI)
___
*[v21.2/troubleshooting-overview.md, line 25 at r4](https://reviewable.io/reviews/cockroachdb/docs/13026#-Mwhj02c3A-ogcwa2CCs:-Mwhj02c3A-ogcwa2CCt:b-fkce36) ([raw file](https://github.com/cockroachdb/docs/blob/91a8193c2fd3d938a318298e995c0a315b581745/v21.2/troubleshooting-overview.md#L25)):*
> ```Markdown
> - If you cannot resolve the issue yourself, the following tools can help you move forward:
>   - [Support Resources](support-resources.html) identify ways you can get help with troubleshooting.
>   - [File an Issue](file-an-issue.html) provides details how to file an issue that you're unable to resolve.
> ```

nit: provides details how -> provides details on how


<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks!

> Reviewed 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13026#-:-MwiYIkP-Gb4lSdxHFbo:b-g2a5fb)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jordanlewis and @vy-ton)



<!-- Sent from Reviewable.io -->

rmloveland reviewed (approved):
LGTM

vy-ton commented on _includes/v21.2/sidebar-data/reference.json:
```diff
@@ -1344,35 +1344,6 @@
         ]
       },
-      {
-        "title": "Performance Optimization",
```
Seems odd to me that these topics, which are core to CockroachDB's architecture/design, would not also be mentioned in `Reference`. Is there an objection to putting them in both `Develop` and `Reference`?

vy-ton reviewed (commented):
@stbof Thanks for tackling this. Sorry for my delayed review. Left some comments about future directions we could take this to better help users.
